independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartCatalog.java

```diff
@@ -165,6 +169,29 @@ public static QuarkusCodestartCatalog fromExtensionsCatalog(ExtensionCatalog cat
         return projectCodestarts;
     }
 
+    private Optional<ExtensionCodestart> getSelectedDefaultCodeStart(QuarkusCodestartProjectInput projectInput) {
+        // This is very hackyish, we need a better data structure to do better
+        boolean is28Or29 = projectInput.getBoms().stream().anyMatch(b -> b.startsWith("io.quarkus:quarkus-bom:2.8.")
```

Comment:
this could possibly work but its really a bad hacky fish (as you said :) - before we +1 this as opposed to change it to the behavior we had before (add resteasy or possibly now resteasy-reactive by default when no codestarts) ... is there something around this that is important for the resteasy-reactive change?

Since I would much rather go with the less optimal but we know works (or at least examples/humans are adjusted to) than this change we risk having to deal with for a long time.

Reply:
You need the hack anyway if you need to default to RESTEasy Reactive for 2.8+

Reply:
Obviously we will make it cleaner in 2.8.1 with a proper version check.

Reply:
FWIW I'm working on a cleaner version.

Reply:
Pushed a new version.

As for enabling RESTEasy Reactive by default, again, we want users to select what they want and not force push a REST impl + examples to them.
Restoring this default behavior will probably break tests and it will take quite some time to figure out all of them especially in the Maven/Gradle tests as they take ages to run.
If you want to pursue this, I'll let you find a guinea pig. I strongly believe it's something we shouldn't do, especially with all the different use cases Quarkus now has.

Reply:
Note that the simple case where people just create a default app still provides a default RESTEasy Reactive app.
It's when you go choose your extensions that you need to make an explicit choice.